Robot-Tests/KW_Leveys.robot

```diff
@@ -0,0 +1,251 @@
+    Vaihda Tietolaji                            ${TL_Leveys_RB}
+    Paikanna osoite                             ${testipaikka}
+    Zoomaa kartta                               5    200
+    Odota sivun latautuminen
+
+    #klikkauksen kokeilu
+    Click Element At Coordinates                ${Kartta}  0  20
+    Wait Until Element Is Visible               ${Leveys_otsikko}
+    Element Should Contain                      ${Leveys_otsikko}    ${Leveys_cm_lukema}
+    Element Should Be Visible                   ${Leveys_lisätty_järjestelmään}
+    Element Should Be Visible                   ${Leveys_muokattu_viimeksi}
+    Element Should Be Visible                   ${Leveys_linkkien_lukumäärä}
+    Element Should Be Visible                   ${Leveys_tietoruutu_boolean}
+    Click Element At Coordinates                ${Kartta}  0  100
+
+    #tuplaklikki
+    Tupla Klikkaa Kartan Keskelle
+    Wait Until Element Is Visible               ${Leveys_otsikko}
+    Element Should Contain                      ${Leveys_otsikko}    ${Leveys_cm_lukema}
+    Element Should Be Visible                   ${Leveys_lisätty_järjestelmään}
+    Element Should Be Visible                   ${Leveys_muokattu_viimeksi}
+    Element Should Be Visible                   ${Leveys_linkkien_lukumäärä}
+    Element Should Be Visible                   ${Leveys_tietoruutu_boolean}
+    Click Element At Coordinates                ${Kartta}  0  100
+
+
+Leveyden_radionappi  [arguments]                ${testipaikka}
+    Log    Testataan radio button
+    Testin Aloitus
+    Vaihda Tietolaji                            ${TL_Leveys_RB}
+    Paikanna osoite                             ${testipaikka}
+    Zoomaa kartta                               5   20 m
+    Odota sivun latautuminen
+    Siirry Muokkaustilaan
+    Wait Until Element Is Not Visible           ${Map_popup}
+
+    Click Element At Coordinates                ${Kartta}  0  20
+    #Annetaan leveystiedot
+    Wait Until Element Is Visible               ${Leveys_muokattava}
+    Select Radio Button                         road-width    enabled
+    Radio Button Should Be Set To               road-width    enabled
+    Wait Until Element Is Visible               ${Leveys_tekstikenttä}
+    Input Text                                  ${Leveys_tekstikenttä}    400    True
+    Click Element                               ${FA_footer_Tallenna}
+    Wait Until Element Is Not Visible           ${Spinner_Overlay}
+
+    #Poistetaan leveystiedot
+    Click Element At Coordinates                ${Kartta}  0  20
+    Wait Until Element Is Visible               ${Leveys_muokattava}
+    Select Radio Button                         road-width    disabled
+    Radio Button Should Be Set To               road-width    disabled
+    Click Element                               ${FA_footer_Tallenna}
+    Wait Until Element Is Not Visible           ${Spinner_Overlay}
+
+
+Leveyden_muokkaus_peruutus  [arguments]         ${testipaikka}
```

Comment:
Tässä testissä voisi mun mielestä olla should-lauseet, jotka osoittaa sen, että leveystieto säilyy muuttumattomana peruutuksen jälkeen.